src/browserContext.ts

```diff
@@ -89,13 +92,16 @@ export abstract class BrowserContextBase extends ExtendedEventEmitter implements
   _browserClosed() {
     for (const page of this.pages())
       page._didClose();
-    this._didCloseInternal();
+    this._didCloseInternal(true);
   }
 
-  _didCloseInternal() {
+  async _didCloseInternal(omitDeleteDownloads = false) {
```

Comment:
This was hard for me to follow. Can we leave it as `_didCloseInternal()`, and add `async deleteDownloads()`. Is it important that the close event fires before the downloads have been deleted?